server/routes/creatingLicences/handlers/confirmCreate.ts

```diff
@@ -11,35 +10,38 @@ import LicenceKind from '../../../enumeration/LicenceKind'
 export default class ConfirmCreateRoutes {
   constructor(
     private readonly communityService: CommunityService,
-    private readonly prisonerService: PrisonerService,
     private readonly licenceService: LicenceService,
     private readonly ukBankHolidayFeedService: UkBankHolidayFeedService
   ) {}
 
   GET = async (req: Request, res: Response): Promise<void> => {
     const { nomisId } = req.params
     const { user } = res.locals
-    const backLink = req.session.returnToCase || '/licence/create/caseload'
+    const backLink = req.session?.returnToCase || '/licence/create/caseload'
 
     const [nomisRecord, deliusRecord, bankHolidays] = await Promise.all([
-      this.prisonerService.getPrisonerDetail(nomisId, user),
+      this.licenceService.getPrisonerDetail(nomisId, user),
       this.communityService.getProbationer({ nomsNumber: nomisId }),
       this.ukBankHolidayFeedService.getEnglishAndWelshHolidays(),
     ])
 
+    if (nomisRecord.cvl.isInHardStopPeriod) {
+      return res.redirect('/access-denied')
+    }
```

Comment:
Unfortunately the hard stop middleware only works when we have a licence object to check `isInHardStopPeriod` on, which we don't have at the point that the middleware runs. Here in the handler however, we have access to the enriched NOMIS record that allows us to check the field. For this reason this route doesn't apply the middleware, but does still prevent access in the hard stop period.

Reply:
add some logging?